be/src/storage/cumulative_compaction.cpp

```diff
@@ -65,62 +81,87 @@ Status CumulativeCompaction::pick_rowsets_to_compact() {
 
     std::sort(candidate_rowsets.begin(), candidate_rowsets.end(), Rowset::comparator);
 
-    RETURN_IF_ERROR(check_version_continuity_with_cumulative_point(candidate_rowsets));
-    RETURN_IF_ERROR(check_version_continuity(candidate_rowsets));
-
     std::vector<RowsetSharedPtr> transient_rowsets;
     size_t compaction_score = 0;
     // the last delete version we meet when traversing candidate_rowsets
     Version last_delete_version{-1, -1};
 
+    int64_t prev_end_version = _tablet->cumulative_layer_point() - 1;
     for (auto rowset : candidate_rowsets) {
+        // meet missed version
+        if (rowset->start_version() != prev_end_version + 1) {
+            if (!transient_rowsets.empty() &&
+                compaction_score >= config::min_cumulative_compaction_num_singleton_deltas) {
+                // min_cumulative_compaction_num_singleton_deltas = 2
+                // 7,6,<4,3>|2-1 -> <4,3>
+                _input_rowsets = transient_rowsets;
+                break;
+            } else {
+                // 7,6,<3>|2-1
+                compaction_score = 0;
+                transient_rowsets.clear();
+            }
+        }
+        // meet a delete version
         if (_tablet->version_for_delete_predicate(rowset->version())) {
             last_delete_version = rowset->version();
-            if (!transient_rowsets.empty()) {
+            if (fit_compaction_condition(transient_rowsets, compaction_score)) {
+                // 7,6,(5),<4,3>|2-1 -> <4,3>
```

Comment:
Is it possible that the rowsets `<4,3>` never satisfy the `fit_compaction_condition`?

Reply:
which scene?

Reply:
What if the `check_version_continuity_with_cumulative_point` is set as a big value?

Reply:
I don't understand what you mean, can you explain more specifically?